File: Semester4/Advanced Programming/ThriftProj/JavaThrift/Lab2_Hw/Networking/src/main/java/objProtocol/SimpleRequest.java
package objProtocol;

import java.io.Serializable;

public class SimpleRequest implements IRequest, Serializable {

    private Object data;
    private RequestType requestType;

    public SimpleRequest(RequestType requestType){
        this.requestType = requestType;
        this.data = null;
    }

    @Override
    public String toString() {
        return "SimpleRequest{" +
                "data=" + data +
                ", requestType=" + requestType +
                '}';
    }

    public SimpleRequest(RequestType requestType, Object data){
        this.requestType = requestType;
        this.data = data;
    }

    @Override
    public Object getData() {
        return data;
    }

    @Override
    public RequestType getRequestType() {
        return this.requestType;
    }
}

File: Semester4/Advanced Programming/ThriftProj/C#Thrift/Lab2_Festival/Common/protocol/SimpleRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Common.protocol
{
    [Serializable]
    public class SimpleRequest : IRequest
    {
        private RequestType requestType;
        private object data;

        public SimpleRequest(RequestType requestType)
        {
            this.requestType = requestType;
            this.data = null;
        }

        public SimpleRequest(RequestType requestType, object data)
        {
            this.requestType = requestType;
            this.data = data;
        }

        public RequestType GetRequestType()
        {
            return this.requestType;
        }

        public object GetData()
        {
            return this.data;
        }

        public override string ToString()
        {
            if (data == null)
                return "Request: {"
                    + " RequestType: " + requestType.ToString() + "}";
                return "Request: {"
                    + " RequestType: " + requestType.ToString()
                    + " Data: " + data.ToString() + "}";
        }
    }
}
